In [15]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import math
import json

In [20]:
data = pd.read_csv("data.csv")

s = str(data["b'radar_primary_yson_msg'"][20])
# with open("cell.json", "w") as file:
#     file.write(s)

print(s)
data


{b'row_step': 1088, b'height': 1, b'fields': [{b'name': b'x', b'datatype': 7, b'count': 1, b'offset': 0}, {b'name': b'y', b'datatype': 7, b'count': 1, b'offset': 4}, {b'name': b'z', b'datatype': 7, b'count': 1, b'offset': 8}, {b'name': b'obj_vrel_long', b'datatype': 7, b'count': 1, b'offset': 12}, {b'name': b'obj_lat_speed', b'datatype': 7, b'count': 1, b'offset': 16}, {b'name': b'obj_rcs_value', b'datatype': 7, b'count': 1, b'offset': 20}, {b'name': b'radial_speed_absolute', b'datatype': 7, b'count': 1, b'offset': 24}, {b'name': b'distance_accuracy', b'datatype': 7, b'count': 1, b'offset': 28}, {b'name': b'angle_accuracy', b'datatype': 7, b'count': 1, b'offset': 32}, {b'name': b'pdh0', b'datatype': 7, b'count': 1, b'offset': 36}, {b'name': b'dist_long_rms', b'datatype': 7, b'count': 1, b'offset': 40}, {b'name': b'dist_lat_rms', b'datatype': 7, b'count': 1, b'offset': 44}, {b'name': b'v_long_rms', b'datatype': 7, b'count': 1, b'offset': 48}, {b'name': b'v_lat_rms', b'datatype': 7, b'co

b'ride_date'  b'rover'  \
0   b'2025-01-23'  b'matar'   
1   b'2025-01-23'  b'matar'   
2   b'2025-01-23'  b'matar'   
3   b'2025-01-23'  b'matar'   
4   b'2025-01-23'  b'matar'   
..            ...       ...   
95  b'2025-01-23'  b'matar'   
96  b'2025-01-23'  b'matar'   
97  b'2025-01-23'  b'matar'   
98  b'2025-01-23'  b'matar'   
99  b'2025-01-23'  b'matar'   

                              b'radar_secondary_list'  \
0   [0, 0, 136, 65, 205, 204, 140, 63, 0, 0, 0, 0,...   
1   [102, 102, 134, 65, 205, 204, 140, 63, 0, 0, 0...   
2   [102, 102, 134, 65, 205, 204, 140, 63, 0, 0, 0...   
3   [102, 102, 134, 65, 205, 204, 140, 63, 0, 0, 0...   
4   [102, 102, 134, 65, 205, 204, 140, 63, 0, 0, 0...   
..                                                ...   
95  [0, 0, 136, 65, 205, 204, 140, 63, 0, 0, 0, 0,...   
96  [0, 0, 136, 65, 205, 204, 140, 63, 0, 0, 0, 0,...   
97  [0, 0, 136, 65, 205, 204, 140, 63, 0, 0, 0, 0,...   
98  [0, 0, 136, 65, 205, 204, 140, 63, 0, 0, 0, 0,...   
99  [205, 204, 8, 66, 51, 51, 227, 192, 0, 0, 0, 0...   

                              b'radar_secondary_data'        b'message_ts'  \
0   [0, 0, 136, 65, 205, 204, 140, 63, 0, 0, 0, 0,...  1737591862924763665   
1   [102, 102, 134, 65, 205, 204, 140, 63, 0, 0, 0...  1737591862996971126   
2   [102, 102, 134, 65, 205, 204, 140, 63, 0, 0, 0...  1737591863073786978   
3   [102, 102, 134, 65, 205, 204, 140, 63, 0, 0, 0...  1737591863144770469   
4   [102, 102, 134, 65, 205, 204, 140, 63, 0, 0, 0...  1737591863213517922   
..                                                ...                  ...   
95  [0, 0, 136, 65, 205, 204, 140, 63, 0, 0, 0, 0,...  1737591869814745323   
96  [0, 0, 136, 65, 205, 204, 140, 63, 0, 0, 0, 0,...  1737591869892168216   
97  [0, 0, 136, 65, 205, 204, 140, 63, 0, 0, 0, 0,...  1737591869959417162   
98  [0, 0, 136, 65, 205, 204, 140, 63, 0, 0, 0, 0,...  1737591870033834600   
99  [205, 204, 8, 66, 51, 51, 227, 192, 0, 0, 0, 0...  1737591870110740282   

                                b'radar_primary_data'  \
0   [0, 0, 136, 65, 205, 204, 140, 63, 0, 0, 0, 0,...   
1   [102, 102, 134, 65, 205, 204, 140, 63, 0, 0, 0...   
2   [102, 102, 134, 65, 205, 204, 140, 63, 0, 0, 0...   
3   [102, 102, 134, 65, 205, 204, 140, 63, 0, 0, 0...   
4   [102, 102, 134, 65, 205, 204, 140, 63, 0, 0, 0...   
..                                                ...   
95  [0, 0, 136, 65, 205, 204, 140, 63, 0, 0, 0, 0,...   
96  [0, 0, 136, 65, 205, 204, 140, 63, 0, 0, 0, 0,...   
97  [0, 0, 136, 65, 205, 204, 140, 63, 0, 0, 0, 0,...   
98  [0, 0, 136, 65, 205, 204, 140, 63, 0, 0, 0, 0,...   
99  [205, 204, 8, 66, 51, 51, 227, 192, 0, 0, 0, 0...   

                          b'radar_secondary_yson_msg'  \
0   {b'row_step': 960, b'height': 1, b'fields': [{...   
1   {b'row_step': 960, b'height': 1, b'fields': [{...   
2   {b'row_step': 1024, b'height': 1, b'fields': [...   
3   {b'row_step': 1024, b'height': 1, b'fields': [...   
4   {b'row_step': 896, b'height': 1, b'fields': [{...   
..                                                ...   
95  {b'row_step': 1024, b'height': 1, b'fields': [...   
96  {b'row_step': 1024, b'height': 1, b'fields': [...   
97  {b'row_step': 960, b'height': 1, b'fields': [{...   
98  {b'row_step': 704, b'height': 1, b'fields': [{...   
99  {b'row_step': 768, b'height': 1, b'fields': [{...   

                              b'/radar/front_primary'  \
0   b'\xbd\x8d\x01\x006\x8c\x91g\x86\x0fL*\x13\x00...   
1   b'\xbe\x8d\x01\x006\x8c\x91g\xc2\xd8\xc4.\x13\...   
2   b'\xbf\x8d\x01\x006\x8c\x91g"\x83=3\x13\x00\x0...   
3   b'\xc0\x8d\x01\x006\x8c\x91g\x82\xf2i7\x13\x00...   
4   b'\xc1\x8d\x01\x006\x8c\x91g\x92s\x98;\x13\x00...   
..                                                ...   
95  b"\x1c\x8e\x01\x00=\x8c\x91g\xaeO\xdb#\x13\x00...   
96  b'\x1d\x8e\x01\x00=\x8c\x91gZ&T(\x13\x00\x00\x...   
97  b'\x1e\x8e\x01\x00=\x8c\x91gRV\x82,\x13\x00\x0...   
98  b'\x1f\x8e\x01\x00=\x8c\x91g\x1eA\xf90\x13\x00...   
99  b' \x8e\x

"{b'row_step': 960, 
b'height': 1,
 b'fields': [{b'name': b'x', b'datatype': 7, b'count': 1,  b'offset': 0}, 
            {b'name': b'y', b'datatype': 7, b'count': 1, b'offset': 4},
             {b'name': b'z', b'datatype': 7, b'count': 1, b'offset': 8}, 
             {b'name': b'obj_vrel_long', b'datatype': 7, b'count': 1, b'offset': 12}, 
             {b'name': b'obj_lat_speed', b'datatype': 7, b'count': 1, b'offset': 16}, 
             {b'name': b'obj_rcs_value', b'datatype': 7, b'count': 1, b'offset': 20}, 
             {b'name': b'radial_speed_absolute', b'datatype': 7, b'count': 1, b'offset': 24}, 
             {b'name': b'distance_accuracy', b'datatype': 7, b'count': 1, b'offset': 28}, 
             {b'name': b'angle_accuracy', b'datatype': 7, b'count': 1, b'offset': 32}, 
             {b'name': b'pdh0', b'datatype': 7, b'count': 1, b'offset': 36}, 
             {b'name': b'dist_long_rms', b'datatype': 7, b'count': 1, b'offset': 40}, 
             {b'name': b'dist_lat_rms', b'datatype': 7, b'count': 1, b'offset': 44}, 
             {b'name': b'v_long_rms', b'datatype': 7, b'count': 1, b'offset': 48}, 
             {b'name': b'v_lat_rms', b'datatype': 7, b'count': 1, b'offset': 52}, 
             {b'name': b'dyn_prop', b'datatype': 1, b'count': 1, b'offset': 56}, 
             {b'name': b'range', b'datatype': 1, b'count': 1, b'offset': 57}, 
             {b'name': b'has_quality', b'datatype': 1, b'count': 1, b'offset': 58}, 
             {b'name': b'ambig', b'datatype': 1, b'count': 1, b'offset': 59}, 
             {b'name': b'invalid', b'datatype': 1, b'count': 1, b'offset': 60}], 
b'data': [0, 0, 136, 65, 205, 204, 140, 63, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 192, 63, 112, 80, 3, 186, 205, 204, 204, 61, 165, 71, 176, 59, 0, 0, 128, 62, 170, 241, 130, 63, 45, 178, 29, 63, 188, 116, 147, 62, 88, 57, 52, 60, 1, 0, 1, 3, 0, 0, 0, 0, 154, 153, 201, 65, 154, 153, 121, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 16, 193, 41, 80, 239, 185, 205, 204, 204, 61, 179, 69, 179, 59, 0, 0, 64, 63, 117, 147, 168, 63, 45, 178, 29, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 0, 0, 12, 66, 51, 51, 227, 192, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 240, 192, 236, 142, 31, 186, 205, 204, 204, 61, 82, 131, 180, 59, 0, 0, 64, 63, 76, 55, 217, 63, 150, 67, 75, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 205, 204, 20, 66, 154, 153, 57, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 160, 192, 58, 166, 1, 186, 205, 204, 204, 61, 65, 230, 176, 59, 0, 0, 64, 63, 117, 147, 168, 63, 45, 178, 29, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 0, 0, 32, 66, 102, 102, 6, 192, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 64, 192, 142, 16, 17, 186, 205, 204, 204, 61, 72, 159, 175, 59, 0, 0, 64, 63, 170, 241, 130, 63, 45, 178, 29, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 102, 102, 42, 66, 0, 0, 96, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 176, 192, 5, 30, 1, 186, 205, 204, 204, 61, 223, 19, 177, 59, 0, 0, 64, 63, 76, 55, 217, 63, 45, 178, 29, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 205, 204, 52, 66, 0, 0, 208, 192, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 192, 192, 149, 84, 26, 186, 205, 204, 204, 61, 254, 0, 179, 59, 0, 0, 64, 63, 76, 55, 217, 63, 150, 67, 75, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 51, 51, 59, 66, 51, 51, 91, 193, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 32, 192, 198, 80, 38, 186, 205, 204, 204, 61, 67, 173, 184, 59, 0, 0, 64, 63, 117, 147, 168, 63, 150, 67, 75, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 51, 51, 67, 66, 102, 102, 170, 193, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 192, 85, 50, 46, 186, 205, 204, 204, 61, 87, 45, 179, 59, 0, 0, 64, 63, 207, 247, 11, 64, 170, 241, 130, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 102, 102, 106, 66, 102, 102, 14, 65, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 176, 192, 8, 21, 240, 185, 205, 204, 204, 61, 194, 51, 179, 59, 0, 0, 64, 63, 186, 73, 52, 64, 170, 241, 130, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 205, 204, 128, 66, 0, 0, 0, 191, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 96, 192, 88, 10, 12, 186, 205, 204, 204, 61, 7, 81, 172, 59, 0, 0, 64, 63, 186, 73, 52, 64, 170, 241, 130, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 154, 153, 139, 66, 205, 204, 108, 192, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 144, 192, 188, 30, 17, 186, 205, 204, 204, 61, 195, 166, 175, 59, 0, 0, 64, 63, 236, 81, 104, 64, 170, 241, 130, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 11, 0, 0, 0, 205, 204, 156, 66, 154, 153, 49, 193, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 191, 27, 31, 26, 186, 205, 204, 204, 61, 202, 242, 178, 59, 0, 0, 64, 63, 207, 247, 11, 64, 117, 147, 168, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 205, 204, 138, 66, 51, 51, 227, 192, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 224, 192, 147, 74, 22, 186, 205, 204, 204, 62, 136, 195, 228, 58, 0, 0, 64, 63, 203, 161, 149, 64, 170, 241, 130, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 1, 1, 3, 4, 0, 0, 0, 0, 0, 7, 67, 0, 0, 176, 192, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 193, 235, 197, 15, 186, 205, 204, 204, 62, 136, 195, 228, 58, 0, 0, 64, 63, 205, 204, 192, 64, 203, 161, 149, 64, 188, 116, 147, 62, 88, 57, 52, 60, 3, 1, 1, 3, 4, 0, 0, 0],
b'width': 15,
b'header': {b'seq': 101821, 
            b'stamp': {b'nsecs': 709627782, b'secs': 1737591862}, 
            b'frame_id': b'radar_front_primary'}, 
b'point_step': 64, 
b'is_bigendian': False, 
b'is_dense': True}"

In [ ]:
s = str(data["b''"][0])

In [17]:
import struct

# Define the data structure
data = bytes([0, 0, 136, 65, 205, 204, 140, 63, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 192, 63, 112, 80, 3, 186, 205, 204, 204, 61, 165, 71, 176, 59, 0, 0, 128, 62, 170, 241, 130, 63, 45, 178, 29, 63, 188, 116, 147, 62, 88, 57, 52, 60, 1, 0, 1, 3, 0, 0, 0, 0, 154, 153, 201, 65, 154, 153, 121, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 16, 193, 41, 80, 239, 185, 205, 204, 204, 61, 179, 69, 179, 59, 0, 0, 64, 63, 117, 147, 168, 63, 45, 178, 29, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 0, 0, 12, 66, 51, 51, 227, 192, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 240, 192, 236, 142, 31, 186, 205, 204, 204, 61, 82, 131, 180, 59, 0, 0, 64, 63, 76, 55, 217, 63, 150, 67, 75, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 205, 204, 20, 66, 154, 153, 57, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 160, 192, 58, 166, 1, 186, 205, 204, 204, 61, 65, 230, 176, 59, 0, 0, 64, 63, 117, 147, 168, 63, 45, 178, 29, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 0, 0, 32, 66, 102, 102, 6, 192, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 64, 192, 142, 16, 17, 186, 205, 204, 204, 61, 72, 159, 175, 59, 0, 0, 64, 63, 170, 241, 130, 63, 45, 178, 29, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 102, 102, 42, 66, 0, 0, 96, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 176, 192, 5, 30, 1, 186, 205, 204, 204, 61, 223, 19, 177, 59, 0, 0, 64, 63, 76, 55, 217, 63, 45, 178, 29, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 205, 204, 52, 66, 0, 0, 208, 192, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 192, 192, 149, 84, 26, 186, 205, 204, 204, 61, 254, 0, 179, 59, 0, 0, 64, 63, 76, 55, 217, 63, 150, 67, 75, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 51, 51, 59, 66, 51, 51, 91, 193, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 32, 192, 198, 80, 38, 186, 205, 204, 204, 61, 67, 173, 184, 59, 0, 0, 64, 63, 117, 147, 168, 63, 150, 67, 75, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 51, 51, 67, 66, 102, 102, 170, 193, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 192, 85, 50, 46, 186, 205, 204, 204, 61, 87, 45, 179, 59, 0, 0, 64, 63, 207, 247, 11, 64, 170, 241, 130, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 102, 102, 106, 66, 102, 102, 14, 65, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 176, 192, 8, 21, 240, 185, 205, 204, 204, 61, 194, 51, 179, 59, 0, 0, 64, 63, 186, 73, 52, 64, 170, 241, 130, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 205, 204, 128, 66, 0, 0, 0, 191, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 96, 192, 88, 10, 12, 186, 205, 204, 204, 61, 7, 81, 172, 59, 0, 0, 64, 63, 186, 73, 52, 64, 170, 241, 130, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 154, 153, 139, 66, 205, 204, 108, 192, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 144, 192, 188, 30, 17, 186, 205, 204, 204, 61, 195, 166, 175, 59, 0, 0, 64, 63, 236, 81, 104, 64, 170, 241, 130, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 11, 0, 0, 0, 205, 204, 156, 66, 154, 153, 49, 193, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 191, 27, 31, 26, 186, 205, 204, 204, 61, 202, 242, 178, 59, 0, 0, 64, 63, 207, 247, 11, 64, 117, 147, 168, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 205, 204, 138, 66, 51, 51, 227, 192, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 224, 192, 147, 74, 22, 186, 205, 204, 204, 62, 136, 195, 228, 58, 0, 0, 64, 63, 203, 161, 149, 64, 170, 241, 130, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 1, 1, 3, 4, 0, 0, 0, 0, 0, 7, 67, 0, 0, 176, 192, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 193, 235, 197, 15, 186, 205, 204, 204, 62, 136, 195, 228, 58, 0, 0, 64, 63, 205, 204, 192, 64, 203, 161, 149, 64, 188, 116, 147, 62, 88, 57, 52, 60, 3, 1, 1, 3, 4, 0, 0, 0])
point_step = 64
width = 15

# Define field offsets (from the 'fields' section)
x_offset = 0
y_offset = 4
z_offset = 8

coordinates = []
for i in range(width):
    start = i * point_step
    x = struct.unpack('<f', data[start + x_offset:start + x_offset + 4])[0]
    y = struct.unpack('<f', data[start + y_offset:start + y_offset + 4])[0]
    z = struct.unpack('<f', data[start + z_offset:start + z_offset + 4])[0]
    coordinates.append((x, y, z))

for idx, (x, y, z) in enumerate(coordinates):
    print(f"Point {idx + 1}: x = {x}, y = {y}, z = {z}")

Point 1: x = 17.0, y = 1.100000023841858, z = 0.0
Point 2: x = 25.200000762939453, y = 3.9000000953674316, z = 0.0
Point 3: x = 35.0, y = -7.099999904632568, z = 0.0
Point 4: x = 37.20000076293945, y = 2.9000000953674316, z = 0.0
Point 5: x = 40.0, y = -2.0999999046325684, z = 0.0
Point 6: x = 42.599998474121094, y = 3.5, z = 0.0
Point 7: x = 45.20000076293945, y = -6.5, z = 0.0
Point 8: x = 46.79999923706055, y = -13.699999809265137, z = 0.0
Point 9: x = 48.79999923706055, y = -21.299999237060547, z = 0.0
Point 10: x = 58.599998474121094, y = 8.899999618530273, z = 0.0
Point 11: x = 64.4000015258789, y = -0.5, z = 0.0
Point 12: x = 69.80000305175781, y = -3.700000047683716, z = 0.0
Point 13: x = 78.4000015258789, y = -11.100000381469727, z = 0.0
Point 14: x = 69.4000015258789, y = -7.099999904632568, z = 0.0
Point 15: x = 135.0, y = -5.5, z = 0.0
